### ISSUE TITLE
GCStress=C test failure: GC\API\GC\GetGeneration\GetGeneration

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/coreclr#15890]
I've seen this several times in the CI.

```
  BEGIN EXECUTION
         "D:\j\workspace\x64_checked_w---0efa51e5\tests\..\bin\tests\Windows_NT.x64.Checked\Tests\Core_Root\corerun.exe" GetGeneration.exe 
        GCStress is on
        ObjectTest Passed!
        0 0
        arrayTest Failed!
        failTest Passed!
        
        Test for GetGeneration() FAILED!
        Expected: 100
        Actual: 1
        END EXECUTION - FAILED
        FAILED
        Test Harness Exitcode is : 1
        
  To run the test:
  > set CORE_ROOT=D:\j\workspace\x64_checked_w---0efa51e5\bin\tests\Windows_NT.x64.Checked\Tests\Core_Root
  > D:\j\workspace\x64_checked_w---0efa51e5\bin\tests\Windows_NT.x64.Checked\GC\API\GC\GetGeneration\GetGeneration.cmd
  
        Expected: True
        Actual:   False
```

Here are two cases:
x64 Windows GCStress=C JitStress=1: https://ci.dot.net/job/dotnet_coreclr/job/master/job/jitstress/job/x64_checked_windows_nt_gcstress0xc_jitstress1_prtest/13/consoleText
x64 Ubuntu GCStress=C: https://ci.dot.net/job/dotnet_coreclr/job/master/job/jitstress/job/x64_checked_ubuntu_gcstress0xc_tst_prtest/23/consoleText

I can't reproduce the failure locally.

**Comment by BruceForstall** [comment]
Is there any reason this test can't work with GCStress?

@swgillespie

**Comment by swgillespie** [comment]
Yeah, if you're doing GCs often enough a new object will get promoted before we get the chance to assert that it's in the expected generation. The test [does try to be robust in the face of GCStress](https://github.com/dotnet/coreclr/blob/master/tests/src/GC/API/GC/GetGeneration.cs#L54) but I'm sure there's a myriad of funny ways that this test can go wrong when GCStress is on.

**Comment by BruceForstall** [comment]
Should we just mark it `<GCStressIncompatible>true</GCStressIncompatible>` and stop running it under GCStress?

**Comment by swgillespie** [comment]
sure!

**Comment by RussKeldorph** [comment]
Fixed by #15894
